File: src/main/java/br/com/kronos/backend/aplicacao/pessoa/Pessoa.java
package br.com.kronos.backend.aplicacao.pessoa;

import static java.util.Objects.requireNonNull;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Pessoa { 
	
	private long id;
	private int numeroRegistro;
	private String email;
	private LocalDate dataNascimento;
	private String nome;
	private String nomeUsual;
	private String nomeSocial;
	private String cpf;
	private boolean bancoTalento;
	private float grauComportamento;
	private Integer idArqAnexo;
	private Integer idCidade;
	private Integer idGenero;
	private Integer idInstituicao;
	
	@Builder
	public Pessoa(long id, int numeroRegistro, String email, LocalDate dataNascimento, String nome, String nomeUsual,
			String nomeSocial, String cpf, boolean bancoTalento, float grauComportamento, Integer idArqAnexo, Integer idCidade,
			Integer idGenero, Integer idInstituicao) {
		super();
		this.id = id;
		this.numeroRegistro = requireNonNull(numeroRegistro, "numeroRegistro é obrigatório");
		this.email = requireNonNull(email, "email é obrigatório");
		this.dataNascimento = requireNonNull(dataNascimento, "dataNascimento é obrigatório");
		this.cpf = requireNonNull(cpf, "cpf é obrigatório");
		this.nome = requireNonNull(nome, "nome é obrigatório");
		this.nomeUsual = requireNonNull(nomeUsual, "nome é obrigatório");
		this.nomeSocial = requireNonNull(nomeSocial, "nomeSocial é obrigatório");
		this.bancoTalento = requireNonNull(bancoTalento, "bancoTalento é obrigatório");
		this.grauComportamento = grauComportamento;
		alterarFoto(idArqAnexo);
		this.idCidade = idCidade;
		this.idGenero = requireNonNull(idGenero, "idGenero é obrigatório");
		this.idInstituicao = requireNonNull(idInstituicao, "idInstitucicao é obrigatório");
	}
	
	public void alterarFoto(Integer idArqAnexo) {
		this.idArqAnexo = idArqAnexo != null && idArqAnexo > 0 ? idArqAnexo : null;
	}

}

File: poc-kronos-frontend/src/model/pessoa/Pessoa.js
class Pessoa {
  constructor(dadosPessoa) {
    this.id = dadosPessoa ? dadosPessoa.id : null;
    this.nome = dadosPessoa ? dadosPessoa.nome : null;
    this.cpf = dadosPessoa ? dadosPessoa.cpf : null;
    this.numeroRegistro = dadosPessoa ? dadosPessoa.numeroRegistro : null;
    this.dataNascimento = dadosPessoa ? dadosPessoa.dataNascimento : null;
    this.email = dadosPessoa ? dadosPessoa.email : null;
    this.idGenero = dadosPessoa ? dadosPessoa.idGenero : null;
    this.idCidade = dadosPessoa ? dadosPessoa.idCidade : null;
    this.idArqAnexo = dadosPessoa ? dadosPessoa.idArqAnexo : null;
    this.nomeUsual = dadosPessoa ? dadosPessoa.nomeUsual : null;
    this.nomeSocial = dadosPessoa ? dadosPessoa.nomeSocial : null;
    this.idInstituicao = dadosPessoa ? dadosPessoa.idInstituicao : null;
    this.bancoTalento = dadosPessoa ? dadosPessoa.bancoTalento : null;
    this.bancoTalentoStr = dadosPessoa
      ? dadosPessoa.bancoTalento === true
        ? "Sim"
        : "Não"
      : null;
    this.grauComportamento = dadosPessoa ? dadosPessoa.grauComportamento : null;
    this.idEstado = null;
  }

  static validations(required, cpf, email, decimal, maxValue) {
    return {
      nome: { required },
      cpf: { required, cpf },
      idCidade: { required },
      idEstado: { required },
      dataNascimento: { required },
      idGenero: { required },
      nomeUsual: { required },
      nomeSocial: { required },
      email: { required, email },
      grauComportamento: { required, decimal, maxValue: maxValue(10.0) }
    };
  }

  static validationMessages() {
    return {
      idEstado: {
        required: "Estado é obrigatório"
      },
      nome: {
        required: "Nome é obrigatório"
      },
      idCidade: {
        required: "Cidade é obrigatório"
      },
      dataNascimento: {
        required: "Data de Nascimento é obrigatório"
      },
      cpf: {
        required: "CPF é obrigatória",
        minLength: "Tamanho do cpf deve ser 11 characters",
        cpf: "Cpf inválido"
      },
      idGenero: {
        required: "Gênero é obrigatório"
      },
      nomeUsual: {
        required: "Nome Usuario é obrigatório"
      },
      nomeSocial: {
        required: "Nome Social é obrigatório"
      },
      email: {
        required: "Email é obrigatório",
        email: "Email inválido"
      },
      grauComportamento: {
        required: "Grau Comportamento é obrigatório",
        decimal: "Somente decimal. Ex: 5.9",
        maxValue: "Valor máximo 10.0"
      }
    };
  }
}
export default Pessoa;
